### ISSUE TITLE
openning file /l/ssd/spill_1_0.log failure: No such file or directory

### ISSUE THREAD

**Issue by nedbass** [LLNL/UnifyCR#74]
I note this error when running `test_write_gotcha`:

```
$ srun -n 1 -N 1 ./test_write_gotcha -f foo -b 1024 -t 1024 -s 1024
rank:0, openning file /l/ssd/spill_0_0.log failure: No such file or directory
This function name failed to be wrapped: stat
Aggregate Write BW is 530.503979MB/s, Min Write BW is 530.503979MB/s
```

**Comment by jgmoore-or** [comment]
The default path for the EXTERNAL_DATA_DIR is hard-coded in unifycr-internal.h to be "/l/ssd".  Setting the UNIFYCR_EXTERNAL_DATA_DIR environment variable to a valid directory prior to invoking the processes fixes this, but it would be better to not rely on hard-coded locations.

**Comment by nedbass** [comment]
@jgmoore-or thanks for the pointers. I'm not sure that explains the error for my case, however, since /l/ssd exists and is writable by me on the system I was running on:

```
$  bass6@catalyst296 /usr/workspace/wsb/bass6/unifycr /dev/pts/0 Thu Jan 04 11:40:03 ((8affa46...)) >
srun -N1 -n1 ./client/tests/test_write_static -s1 -b1 -t1 -f /tmp/foo
rank:0, openning file /l/ssd/spill_0_0.log failure
srun: error: catalyst296: task 0: Floating point exception (core dumped)
$  bass6@catalyst296 /usr/workspace/wsb/bass6/unifycr /dev/pts/0 Thu Jan 04 11:40:08 ((8affa46...)) >
ls -l /l/ssd
total 0
-rw------- 1 bass6 bass6  0 Jan  4 11:40 serverdbg0.log
-rw------- 1 bass6 bass6  0 Jan  4 11:40 spill_index_0_0.log
drwx------ 2 bass6 bass6 85 Jan  4 11:40 unifycr_db-0-0/
drwx------ 2 bass6 bass6 85 Jan  4 11:40 unifycr_db-0-0_stats/
drwx------ 2 bass6 bass6 85 Jan  4 11:40 unifycr_db-1-0/
drwx------ 2 bass6 bass6 85 Jan  4 11:40 unifycr_db-1-0_stats/
```

**Comment by nedbass** [comment]
I get this error when I don't set UNIFYCR_USE_SPILLOVER=1. This also seems somehow connected to #76.

**Comment by jgmoore-or** [comment]
Ned,

Sorry, I thought it was due to directory path not being there.  I hit this
too, and should have entered a defect:

... the spill file gets created by the client library only if the
environment variable, UNIFYCR_USE_SPILLOVER is set to a non-zero
value.  The server assumes this file is created always, but the client only
creates it if the variable is set.

I think our best option is to make UNIFYCR_USE_SPILLOVER the default
behavior in the client, since the server won't work if it's not set.

**Comment by dsikich** [comment]
@nedbass @jgmoore-or I think this one can be closed now. 

If the external meta or data dirs are not set the client will fail with a message that they were not set properly. The spillover default is now =1, so you shouldn't have to set it anymore. 

#87 #88

**Comment by MichaelBrim** [comment]
This should be closed when PR #132 is merged.
